### PR TITLE
integration, benches, algebra: Fixup tests and benchmarks in generic refactor

### PR DESCRIPTION
**Pull request by joeykraut**
### Purpose
This PR brings all tests, clippy lints, benchmarks, etc into a working state after refactoring the crate to be generic over curve choice. 

### Testing
- Unit and integration tests pass

### CODE REVIEW

File: src/algebra/scalar.rs [hunk 1]
```diff
@@ -65,11 +66,7 @@ impl<C: CurveGroup> Scalar<C> {
     ///
     /// TODO: Validate that this gives a uniform distribution over the field
```
Comment (akirillo) [inline src/algebra/scalar.rs:67]: nit: do we still need this `TODO` comment? we are using `UniformRand` now lol

**Review by akirillo — APPROVED** [review]
couple nits/questions but nothing blocking

File: src/lib.rs [hunk 2]
```diff
@@ -58,12 +58,16 @@ pub mod test_helpers {
...
-        algebra::test_helper::TestCurve,
         beaver::PartyIDBeaverSource,
         network::{MockNetwork, NoRecvNetwork, UnboundedDuplexStream},
         MpcFabric, PARTY0, PARTY1,
     };
 
+    use ark_curve25519::EdwardsProjective as Curve25519Projective;
+
+    /// A curve used for testing algebra implementations, set to curve25519
+    pub type TestCurve = Curve25519Projective;
```
[the remark this answers is not in the document]
Comment (joeykraut) [reply src/lib.rs:69]: It's redefined outside of the crate, i.e. in integration tests which is fine

File: src/algebra/curve.rs
[the remark this answers is not in the document]
Comment (joeykraut) [reply src/algebra/curve.rs:1]: Yeah, we have no reference implementation to compare against (because it's now generic), so the tests have changed

**Pull request merged**